lib/src/scss-shared/_nx-form-validation.scss

```diff
@@ -5,20 +5,70 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 @import 'nx-variables';
+@import 'nx-container-helpers';
 
 @mixin valid() {
-  border-left-color: $nx-valid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-valid-color;
+  }
+
+  .nx-text-input__validation-icon {
+    display: inline-block;
+
+    .nx-icon--valid {
+      display: block;
+    }
+  }
 }
 
 @mixin invalid() {
-  border-color: $nx-invalid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-invalid-color;
+  }
+
+  > .nx-text-input__invalid-message {
+    visibility: visible;
+  }
+
+  .nx-text-input__validation-icon {
+    display: inline-block;
+
+    .nx-icon--invalid {
+      display: block;
+    }
+  }
 }
 
 @mixin pristine() {
-  border-color: $nx-grey-4;
-  border-left-color: $nx-valid-color;
-  border-left-style: double;
-  color: $nx-text-color-light;
+  .nx-text-input__input:not(:disabled) {
+    border-color: #d1d6f0;
+    color: $nx-text-color-light;
+  }
+}
+
+.nx-text-input__validation-icon {
+  &, .nx-icon {
+    display: none;
+  }
+}
+
+.nx-text-input__invalid-message {
+  color: #cd0028;
+  font-size: $nx-font-size-xs;
+  margin-top: $nx-spacing-xs;
+  visibility: hidden;
+}
+
+.nx-icon--invalid {
+  color: #cd0028;
+  height: 1em;
+  margin-left: 0;
+  margin-right: $nx-spacing-md;
```

Comment:
Tieing into my comment above: "I feel like this padding should probably be on the box; with the input itself having none" this margin goes against our standard margin and padding rules.  `nx-text-input__box` should use the `container-horizontal` mixin to negate undesireable default margins on the icons.

Reply:
The problem with that is that because they are inside a container themselves the first icon gets stripped of its left margin, but keeps its right, and the opposite happens with the second icon, but we don't actually want either icon to have any l/r margin